### PR TITLE
Bs and es major

### PR DESCRIPTION
**Pull request by neketka**
### Summary 

Added biology and society major, added environment & sustainability major without concentrations

Remaining TODOs:

- [ ] Add concentrations.

### Test Plan 

Check that the appropriate requirements are fulfilled.

### CODE REVIEW

File: src/requirements/data/majors/bsoc.ts [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+      'PHIL 3410',
+      'STS 2051',
+      'STS 2061'
+    ),
+    fulfilledBy: 'courses',
+    perSlotMinCount: [1],
+    slotNames: ['Course'],
+  },
+  {
+    name: 'Social sciences/humanities foundation',
```
Comment (jason-tung) [inline src/requirements/data/majors/bsoc.ts:65]: capitalize the name

File: src/requirements/data/majors/bsoc.ts [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+    slotNames: [
+      'BIOG 1440 or BIOG 1445',
+      'BIOEE 1610 or BIOSM 1610',
+      'BIOMG 1350',
+      'BIOEE 1780 or BIOSM 1780',
+    ],
+  },
+  {
+    name: 'College Calculus',
+    description: 'College calculus is required for Biology & Society Majors',
```
Comment (jason-tung) [inline src/requirements/data/majors/bsoc.ts:29]: add period to end

File: src/requirements/data/majors/bsoc.ts [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+      'STS 2051',
+      'STS 2061'
+    ),
+    fulfilledBy: 'courses',
+    perSlotMinCount: [1],
+    slotNames: ['Course'],
+  },
+  {
+    name: 'Social sciences/humanities foundation',
+    description: 'Two courses; one from any two of the following subject areas',
```
Comment (jason-tung) [inline src/requirements/data/majors/bsoc.ts:66]: add period

File: src/requirements/data/majors/bsoc.ts [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+        'STS 3111',
+      ],
+      ['BSOC 3311', 'PAM 2350', 'PAM 2780', 'STS 3241', 'STS 3911'],
+      ['COMM 3210', 'STS 2851', 'STS 3020'],
+      ['COML 2035', 'ENGL 2780']
+    ),
+    fulfilledBy: 'courses',
+    perSlotMinCount: [1, 1, 1, 1, 1, 1],
+    slotNames: [
+      'History of science',
```
Comment (jason-tung) [inline src/requirements/data/majors/bsoc.ts:100]: History of science -> History of Science

File: src/requirements/data/majors/bsoc.ts [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+      'Philosophy of science',
```
Comment (jason-tung) [inline src/requirements/data/majors/bsoc.ts:101]: same for all of these

File: src/requirements/data/majors/bsoc.ts [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+      'Sociology of science',
+      'Politics of science',
+      'Science communication',
+      'Literature and science',
+    ],
+    minNumberOfSlots: 2,
+  },
+  {
+    name: 'Biology foundation (breadth)',
```
Comment (jason-tung) [inline src/requirements/data/majors/bsoc.ts:110]: Biology Foundation (Breadth)

File: src/requirements/data/majors/bsoc.ts [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+    description: 'Three courses; one each from three of the following subject areas',
```
Comment (jason-tung) [inline src/requirements/data/majors/bsoc.ts:111]: end sentence with period

File: src/requirements/data/majors/bsoc.ts [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+        'PLHRT 4730',
+      ],
+      ['BIOMG 2800', 'NTRES 2830', 'PLBRG 2250'],
+      ['BIONB 2220'],
+      ['NS 1150', 'NS 1220']
+    ),
+    fulfilledBy: 'courses',
+    perSlotMinCount: [1, 1, 1, 1, 1, 1, 1, 1],
+    slotNames: [
+      'Anatomy and physiology',
```
Comment (jason-tung) [inline src/requirements/data/majors/bsoc.ts:155]: caps here too

File: src/requirements/data/majors/bsoc.ts [hunk 9]
```diff
@@ -0,0 +1,320 @@
...
+      'Ecology',
+      'Genetics and development',
+      'Neurobiology',
+      'Nutrition',
+    ],
+  },
+  {
+    name: 'Biology foundation (Depth requirement)',
+    description:
+      'One biology course for which one of the above (2c) is a prerequisite. (Contact dept for list of approved nutrition depth courses)',
```
Comment (jason-tung) [inline src/requirements/data/majors/bsoc.ts:168]: period

File: src/requirements/data/majors/bsoc.ts [hunk 10]
```diff
@@ -0,0 +1,320 @@
...
+    source: 'https://courses.cornell.edu/preview_program.php?catoid=41&poid=19803',
+    checker: includesWithSingleRequirement('BSOC 1942', 'BSOC 2468', 'BSOC 3011', 'NTRES 2320'),
+    fulfilledBy: 'courses',
+    perSlotMinCount: [1],
+    slotNames: ['Core Course'],
+  },
+  {
+    name: 'Theme',
+    description:
+      '(Five courses that correspond to the theme selected by the student). These courses must be above the 1000-level, at least 3 credit hours, and taken for a letter grade. For a course to count towards the major, students must receive at least a C- as a final grade.',
```
Comment (jason-tung) [inline src/requirements/data/majors/bsoc.ts:203]: awk to start desc with parens

File: src/requirements/data/majors/ess.ts [hunk 11]
```diff
@@ -0,0 +1,76 @@
+import { CollegeOrMajorRequirement } from '../../types';
+import { includesWithSubRequirements } from '../checkers-common';
+
+const environmentAndSustainability: readonly CollegeOrMajorRequirement[] = [
+  {
+    name: 'Required courses',
+    description: 'Required courses',
```
Comment (jason-tung) [inline src/requirements/data/majors/ess.ts:7]: caps

File: src/requirements/data/majors/ess.ts [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+      'Social science',
+      'Biology',
+      'Chemistry/physics',
+      'Statistics',
+      'Humanities',
+      'Economics',
+      'Field/Engaged experience',
+      'Sustainability science colloquium',
+      'Capstone course',
+    ],
```
Comment (jason-tung) [inline src/requirements/data/majors/ess.ts:63]: caps

**Review by jason-tung — CHANGES_REQUESTED** [review]
Looks mostly fine! Just some minor issues with formatting

File: src/requirements/data/majors/bsoc.ts [hunk 13]
```diff
@@ -0,0 +1,309 @@
...
+    description: 'One course (should be completed by end of junior year).',
+    source: 'https://sts.cornell.edu/bio-society#requirements',
+    checker: includesWithSingleRequirement('BSOC 2051', 'BSOC 2061'),
+    fulfilledBy: 'courses',
+    perSlotMinCount: [1],
+    slotNames: ['BSOC 2051 or 2061'],
+  },
+  {
+    name: 'Social Sciences/Humanities Foundation',
+    description: 'Two courses; one from any two of the following subject areas',
```
Comment (tcho6319) [inline src/requirements/data/majors/bsoc.ts:70]: Nit: add period

File: src/requirements/data/majors/ess.ts [hunk 14]
```diff
@@ -0,0 +1,641 @@
+import { CollegeOrMajorRequirement } from '../../types';
+import { includesWithSingleRequirement, includesWithSubRequirements } from '../checkers-common';
+
+const environmentAndSustainability: readonly CollegeOrMajorRequirement[] = [
+  {
+    name: 'Foundation Course',
+    description: 'Choose one course',
```
Comment (tcho6319) [inline src/requirements/data/majors/ess.ts:7]: Nit: Add periods to end of descriptions

File: src/requirements/data/majors/ess.ts [hunk 15]
```diff
@@ -0,0 +1,641 @@
...
+          'NTRES 3330',
+          'RELST 2273',
+          'STS 3181',
+          'STS 4131',
+          'STS 4460',
+          'PHIL 1440'
+        ),
+        perSlotMinCount: [7],
+        minNumberOfSlots: 1,
+        slotNames: ['Courses'],
```
Comment (tcho6319) [inline src/requirements/data/majors/ess.ts:215]: Nit: singular slotName 'Course'

File: src/requirements/data/majors/ess.ts [hunk 16]
```diff
@@ -0,0 +1,641 @@
...
+        perSlotMinCount: [1, 1, 1, 1, 1, 2, 1],
+        slotNames: [
+          'Course',
+          'Course',
+          'Course',
+          'Course',
+          'Course',
+          'Resource/Environmental Economics',
+          'Data Analysis/Econometrics',
+        ],
```
Comment (tcho6319) [inline src/requirements/data/majors/ess.ts:178]: Nit: To be consistent with the slotNames of the other concentrations, should we include the group names e.g. 'A. Required Course', 'B. Resource/Environmental Economics', 'C. Data Analysis/Econometrics'?

**Review by tcho6319 — APPROVED** [review]
Overall LGTM but I left a few nits! @einc you should squash and merge when you're ready.

File: src/requirements/data/majors/ess.ts (hunk 16 above)
Comment (einc) [reply src/requirements/data/majors/ess.ts:178]: yes, added these!

**Review by tcho6319 — COMMENTED** [review]
@einc Changes look good to me but you should make sure to re-run the tests and re-generate the decorated requirements json because your changes don't show up on the PR's staging site.

**Review by jason-tung — APPROVED** [review]

**Pull request merged**